### PR TITLE
Calculate and display modification areas

### PR DESCRIPTION
**Pull request by caseypt**
* Calculate area of modification geojson and replace placeholder value in
modification summary with actual value.

To test:

- Create a project.
- Draw some modifications.
- Check that areas are computed in the modifications summary.

selection_040

Connects to #236

### CODE REVIEW

File: src/mmw/js/src/modeling/models.js [hunk 1]
```diff
@@ -102,6 +105,18 @@ var ModificationModel = Backbone.Model.extend({
...
+    initialize: function() {
+        this.setAcres();
+        this.on('change:geojson', this.setAcres);
+    },
+
+    setAcres: function() {
+        var areaInMeters = turfArea(this.get('geojson')),
+            areaInAcres = areaInMeters * 0.000247105;
+
+        this.set('area_acres', areaInAcres);
```
Comment (kdeloach) [inline src/mmw/js/src/modeling/models.js]: Consider adding a default value for this attribute.

File: src/mmw/js/src/modeling/models.js [hunk 2]
```diff
@@ -38,7 +39,9 @@ var ProjectModel = Backbone.Model.extend({
         this.set('model_package', 'tr-55');
         this.set('taskModel', new Tr55TaskModel());
 
-        this.makeFirstScenarioActive();
+        // After the scenario collection is initialized, set
+        // the first scenario as the active scenario.
+        this.listenToOnce(this, 'change:scenarios', this.makeFirstScenarioActive);
```
Comment (kdeloach) [inline src/mmw/js/src/modeling/models.js]: I was planning on treating modifications as immutable so we probably don' need to listen for property changes. We could just set `area_acres` once upon initialization.

**Comment by rajadain** [comment]
There seem to be a lot of unrelated changes to `npm-shrinkwrap.json`. Are we sure those are not accidental?

File: src/mmw/js/src/modeling/models.js (hunk 2 above)
Comment (caseypt) [inline src/mmw/js/src/modeling/models.js]: You mean this for line 112, correct?

File: src/mmw/js/src/modeling/models.js (hunk 1 above)
Comment (caseypt) [inline src/mmw/js/src/modeling/models.js]: Good idea.

**Comment by caseypt** [comment]
I followed the workflow [here](https://docs.npmjs.com/cli/shrinkwrap#building-shrinkwrapped-packages) for updating the shrinkwrap. Have we been following another process?

File: src/mmw/js/src/modeling/models.js (hunk 2 above)
Comment (kdeloach) [inline src/mmw/js/src/modeling/models.js]: Yes.

**Comment by rajadain** [comment]
Those instructions seem like the ones we've been following. @kdeloach's ab6f509 and b2845e1 had only specific changes to the file, but his 8b7faac changed the whole file, so perhaps this isn't as unusual as I'd assumed.

**Comment by caseypt** [comment]
Reworked to use sq ft for areas smaller than an acre and acres for anything larger.

To test:

- Draw some modifications.
- Verify that the ones measured in sq ft are less then 43,560 sq ft (the number of sq ft in an acre).

File: src/mmw/js/src/modeling/models.js [hunk 3]
```diff
@@ -101,7 +104,27 @@ var ModificationModel = Backbone.Model.extend({
     defaults: {
         name: '',
         type: '',
-        geojson: null
+        geojson: null,
+        area: '0',
+        units: 'meter(s)'
```
Comment (caseypt) [inline src/mmw/js/src/modeling/models.js]: @kdeloach defaults set.

File: src/mmw/js/src/modeling/models.js [hunk 4]
```diff
@@ -101,7 +104,27 @@ var ModificationModel = Backbone.Model.extend({
...
+    },
+
+    initialize: function() {
+        this.setDisplayArea();
```
Comment (caseypt) [inline src/mmw/js/src/modeling/models.js]: @kdeloach event removed.

**Comment by kdeloach** [comment]
RE: npm shrinkwrap -- I like to treat updating the shrinkwrap like whitespace or EOL changes. Here is the unofficial workflow I've been following.

1. Run `npm shrinkwrap`
2. Commit (if there were no changes-- great!)
3. Run `npm install foo --save`
4. Run `npm shrinkwrap` (again)
5. Commit

**Comment by kdeloach** [comment]
Consider adding number formatting (commas)?
image

**Comment by kdeloach** [comment]
+1 Works great

**Comment by caseypt** [comment]
Updated with some tests, and a custom template filter to add commas to the modification areas:

selection_042

**Comment by mmcfarland** [comment]
+1 Looks good.

**Pull request merged**